documentation/modules/ROOT/pages/connectors/db2.adoc

```diff
@@ -0,0 +1,1503 @@
+However these are only for convenience and manually performing the same commands using
+db2 control commands would have the same effect. 
+
+We assume here that the contents of 
+debezium-connector-db2/src/test/docker/db2-cdc-docker are available
+on the machine on which Db2 is running in $HOME/asncdctools/src
+and that the user has logged-in as the db2inst1 user.
+
+First compile this function on the Db2 server, using the _bldrtn_
+C-compiler
+
+[source,shell]
+
+----
+$ cd $HOME/asncdctools/src
+----
+
+[source,shell]
+
+----
+$ ./bldrtn asncdc
+----
+
+
+Start the database if not already running:
+
+
+[source,shell]
+
+----
+$ db2 start db <DB NAME>
+
+----
+
+We need to make sure that the metadata catalog can be read via JDBC
+
+[source,shell]
+----
+$ cd $HOME/sqllib/bnd
+----
+
+[source,shell]
+----
+$ db2 bind db2schema.bnd blocking all grant public sqlerror continue 
+----
+
+The database needs to have been recently backed-up so doing following if this is not the case:
+
+[source,shell]
+----
+$ db2 backup db <DB NAME> to /dev/null
+----
+
+[source,shell]
+----
+$ db2 restart db <DB NAME>
+----
+
+Install the UDF:
+
+[source,shell]
+
+----
+$ db2 connect to <DB NAME>
+----
+
+We assume that the db2 is installed on the db2inst1 user
+
+[source,shell]
+----
+$ cp $HOME/asncdctools/src/asncdc $HOME/sqllib/function
+----
+
+[source,shell]
+----
+$ chmod 777 $HOME/sqllib/function
+
+----
+
+
+
+Enable the UDF that allows the ASN capture agent to be started/stopped
+
+[source,shell]
+
+----
+$ db2 -tvmf $HOME/asncdctools/src/asncdc_UDF.sql
+----
+
+Create the ASN Control tables
+
+[source,shell]
+
+----
+$ db2 -tvmf $HOME/asncdctools/src/asncdctables.sql
+----
+
+
+Enable the  UDF that allows us to add/remove tables to be captured
+
+[source,shell]
+
+----
+$ db2 -tvmf $HOME/asncdctools/src/asncdcaddremove.sql
+----
+
+
+
+Having done the above, we can use the UDFs to control ASN via SQL commands.
+Some of the UDFs expect a return value in which case we use the SQL _VALUE_
+statement to invoke them, while others are fire-and-forget in which case we
+use the SQL _CALL_ statement.
+
+
+First the ASN agent needs to be started:
+
+[source,sql]
+----
+VALUES ASNCDC.ASNCDCSERVICES('start','asncdc');
+----
+
+If the agent every needs to be stopped:
+
+[source,sql]
+----
+VALUES ASNCDC.ASNCDCSERVICES('stop','asncdc');
+----
+
+The status if the agent can be checked at any moment:
+
+[source,sql]
+----
+VALUES ASNCDC.ASNCDCSERVICES('status','asncdc');
+----
+
+
+A table MYTABLE in MYSCHEMA can be put into capture mode by doing:
+
+[source,sql]
+----
+CALL ASNCDC.ADDTABLE('MYSCHEMA', 'MYTABLE' )";
+----
+
+
+At table MYTABLE in MYSCHEMA can be removed from capture mode by doing:
+
+
+[source,sql]
+----
+CALL ASNCDC.REMOVETABLE('MYSCHEMA', 'MYTABLE' );
+----
+
+After a table is added or removed, the user __MUST__ reinitialize the ASN service:
+
+
+[source,sql]
+----
+VALUES ASNCDC.ASNCDCSERVICES('reinit','asncdc');
+----
+
+== How the Db2 connector works
```

Comment:
Is there mention of downloading the JDBC driver and adding it to the contents of the connector archive somewhere?

Reply:
We have a dependency in the pom file.
We don't explicitly download the jdbc driver.
We assume the release process when the generating the jars pulls the dependencies.
What do we need to add in the documentation ?

Reply:
Debezium is typically not consumed as a Maven dependency (the [embedded mode](https://debezium.io/documentation/reference/1.0/operations/embedded.html) being the exception)), but as a plug-in that's added to Kafka Connect. Those plug-ins are obtained as ZIP/TAR.GZ archives from Maven Central. In case of DB2, we cannot add the driver JAR to that connector archive due to its license (unlike as for most other connectors), so a user must

* obtain and extract the DB2 connector archive
* obtain the JDBC driver and add it into the extraced connector directory

That's what I started the concurrent discussion with Erin for, trying to gauge whether we actually can have this changed so we can redistribute the driver in the connector archive.

Reply:
Ok. Sorry I misunderstood what you meant.
Done.